src/test/java/chalmers/pimp/model/canvas/layer/DoodleTest.java

```diff
@@ -0,0 +1,235 @@
+package chalmers.pimp.model.canvas.layer;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import chalmers.pimp.model.IRenderer;
+import chalmers.pimp.model.color.ColorFactory;
+import chalmers.pimp.model.color.IColor;
+import chalmers.pimp.model.color.IReadOnlyColor;
+import chalmers.pimp.model.pixeldata.IPixel;
+import chalmers.pimp.model.pixeldata.IReadOnlyPixelData;
+import chalmers.pimp.model.pixeldata.PixelFactory;
+import java.awt.Component;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class DoodleTest {
+
+  static class Line {
+
+    private final int x1, y1, x2, y2;
+
+    Line(int x1, int y1, int x2, int y2) {
+      this.x1 = x1;
+      this.y1 = y1;
+      this.x2 = x2;
+      this.y2 = y2;
+    }
+
+    Line(IPixel p1, IPixel p2) {
+      this.x1 = p1.getX();
+      this.y1 = p1.getY();
+      this.x2 = p2.getX();
+      this.y2 = p2.getY();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (!(o instanceof Line)) {
+        return false;
+      }
+
+      Line l = (Line) o;
+
+      return (this.x1 == l.x1 && this.y1 == l.y1 && this.x2 == l.x2 && this.y2 == l.y2) || (
+          this.x1 == l.x2 && this.y1 == l.y2 && this.x2 == l.x1 && this.y2 == l.y1);
+    }
+  }
+
+  static class RendererMock implements IRenderer {
```

Comment:
This is a handy class for testing! You can extract it to the same level as the test class so that it can be used by other tests more easily.